injector/src/main/java/edu/ucr/cs/riple/injector/Helper.java

```diff
@@ -608,7 +609,14 @@ public static Range findSimpleNameRangeInTypeName(Type type) {
       return ((ClassOrInterfaceType) type).getName().getRange().get();
     }
     if (type instanceof ArrayType) {
-      return findSimpleNameRangeInTypeName(((ArrayType) type).getComponentType());
+      Optional<TokenRange> tokenRange = type.getTokenRange();
+      int index = type.toString().indexOf('[');
```

Comment:
This is not the right way to compute the index. The `type.toString()` will return string representation of the type, not how it is written in the source code. You should find the index via:
```
for (JavaToken javaToken : range) {
   // Token T = javaToken.asString();
   // Your logic
}
```

Reply:
Updated to this logic

Reply:
@avenger2597 Nice, this is much cleaner now.